src/ser/mod.rs

```diff
@@ -361,6 +357,46 @@ mod tests {
         assert_eq!(got, should_be);
     }
 
+    #[test]
+    fn test_serialize_simple_map() {
+        let mut hashmap = std::collections::HashMap::new();
+        hashmap.insert("key1", "val1");
+        let mut buffer = Vec::new();
+        {
+            let mut ser = Serializer::new(&mut buffer);
+            hashmap
+                .serialize(&mut ser)
+                .expect("unable to serialize a hashmap instance");
+        }
+        let got = String::from_utf8(buffer).unwrap();
+        assert_eq!("<key1>val1</key1>", got)
+    }
+
+    #[test]
+    fn test_serialize_map_within_struct() {
```

Comment:
This test actually fails intermittently. 

I'm wondering if it is from the implementation of serializing Map's or the test [relying](https://github.com/RReverser/serde-xml-rs/pull/154/files#diff-d6adfba5b9892574146927e3f707649f3b601313211502ff5ae3c796f83dcee4R395) on a fixed order of printed keys.

Reply:
Exactly, this test is relying on the Map's items being ordered, which is totally wrong. Sorry for that.

I'd suggest removing the second insert or relaxing the assert using `.contains()` to assert that at least the content is ok, rather than the exact output content+order.

Reply:
Thanks for the hints, I pushed an update to fix the test.

Reply:
> Exactly, this test is relying on the Map's items being ordered, which is totally wrong. Sorry for that.

ah, makes sense @adrianbenavides. Thank you for explaining! No need to apologize too. Part of learning!

Reply:
stellar, thanks @joemat!

Reply:
Huh, cannot find the resolve button. I ran `cargo test ser` repeatedly locally and things look great!